### PR TITLE
Feature/redesign auth

### PR DESCRIPTION
**Pull request by eadm**
**YouTrack task**: [#APPS-1459](https://vyahhi.myjetbrains.com/youtrack/issue/APPS-1459)

**Description List**:
* New design for LaunchActivity, LoginActivity and RegisterActivity
* Screenshots available at corresponding YouTrack tasks

### CODE REVIEW

File: app/src/main/java/org/stepic/droid/ui/activities/LoginActivity.kt [hunk 1]
```diff
@@ -166,28 +187,38 @@ class LoginActivity : FragmentActivityBase(), LoginView {
...
 
+    private fun onClearLoginError() {
+        loginButton.isEnabled = true
+        loginForm.isEnabled = true
+        loginErrorMessage.visibility = View.GONE
+    }
+
     override fun onFailLogin(type: LoginFailType, credential: Credential?) {
-        Toast.makeText(this, getMessageFor(type), Toast.LENGTH_SHORT).show()
+        loginForm.isEnabled = false
```
Comment (KirillMakarov) [inline app/src/main/java/org/stepic/droid/ui/activities/LoginActivity.kt]: user can't resubmit login+password, when internet connection is lost. It is not right behaviour

File: app/src/main/java/org/stepic/droid/util/TextViewUtil.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        val end = spannable.getSpanEnd(it)
+        val flags = spannable.getSpanFlags(it)
+        spannable.removeSpan(it)
+        spannable.setSpan(URLSpanWithoutUnderline(it.url), start, end, flags)
+    }
+    tv.text = spannable
+}
+
+private class URLSpanWithoutUnderline(url: String) : URLSpan(url) {
+    override fun updateDrawState(ds: TextPaint?) {
```
Comment (KirillMakarov) [inline app/src/main/java/org/stepic/droid/util/TextViewUtil.kt]: what is ds?

File: app/src/classic/java/org.stepic.droid/ui/activities/LaunchActivity.kt [hunk 3]
```diff
@@ -172,34 +180,52 @@ class LaunchActivity : BackToExitActivityBase(), LoginView {
...
-        val display = windowManager.defaultDisplay
-        val size = Point()
-        display.getSize(size)
-        val widthOfScreen = size.x
+    private fun initSocialRecycler(googleApiClient: GoogleApiClient?, state: SocialAuthAdapter.State = SocialAuthAdapter.State.NORMAL) {
+        socialListRecyclerView.layoutManager = GridLayoutManager(this, 3)
 
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+            socialListRecyclerView.itemAnimator = FadeInDownAnimator()
+            socialListRecyclerView.itemAnimator.removeDuration = 0
```
Comment (KirillMakarov) [inline app/src/classic/java/org.stepic.droid/ui/activities/LaunchActivity.kt]: why remove duration is zero, it will broke "show less" animation.

**Comment by KirillMakarov** [comment]
It is Sign in screen on nexus 5.
It looks bad:
* sign in button is not shown
* keyboard is shown after user has open screen. We should hide it by default.
* the color of marker is default (use new_accent_color)
image

**Comment by KirillMakarov** [comment]
please, remove slide in - slide out animation on the screens

File: app/src/main/res/values/styles.xml [hunk 4]
```diff
@@ -111,7 +111,10 @@
         <item name="android:windowBackground">@color/login_launch_registration_background</item>
         <item name="android:colorBackground">@color/login_launch_registration_background</item>
         <item name="android:textColor">@color/text_color</item>
-        <item name="android:statusBarColor">@color/new_accent_color</item>
+        <item name="android:statusBarColor" tools:targetApi="lollipop">@color/new_accent_color</item>
```
Comment (KirillMakarov) [inline app/src/main/res/values/styles.xml]: `colorPrimaryDark` is used for status bar color, we _can_ use it

File: app/src/main/res/values/styles.xml [hunk 5]
```diff
@@ -246,6 +249,7 @@
         <item name="android:maxLines">1</item>
         <item name="android:background">@android:color/transparent</item>
         <item name="android:textCursorDrawable">@null</item>
+        <item name="android:textColorHighlight">@color/drawer_item_highlighting_pressed</item>
```
Comment (KirillMakarov) [inline app/src/main/res/values/styles.xml]: The drawer is not used anymore, but this color is ok for highlighting. In future we should rename this color.

File: app/src/classic/java/org.stepic.droid/ui/activities/LaunchActivity.kt [hunk 6]
```diff
@@ -172,34 +179,47 @@ class LaunchActivity : BackToExitActivityBase(), LoginView {
...
+    private fun initSocialRecycler(googleApiClient: GoogleApiClient?, state: SocialAuthAdapter.State = SocialAuthAdapter.State.NORMAL) {
+        socialListRecyclerView.layoutManager = GridLayoutManager(this, 3)
 
-        val display = windowManager.defaultDisplay
-        val size = Point()
-        display.getSize(size)
-        val widthOfScreen = size.x
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+            socialListRecyclerView.itemAnimator = FadeInDownAnimator()
+            socialListRecyclerView.itemAnimator.removeDuration = 0
```
Comment (KirillMakarov) [inline app/src/classic/java/org.stepic.droid/ui/activities/LaunchActivity.kt]: collapse animation is not working

Comment (eadm) [reply app/src/classic/java/org.stepic.droid/ui/activities/LaunchActivity.kt]: collapse animation not working properly so I disabled it

**Pull request merged**